app/models/concerns/orchestration/dhcp.rb

```diff
@@ -82,25 +82,15 @@ def boot_server
     # if we don't manage tftp for IPv4 at all, we dont create a next-server entry.
     return unless tftp?
 
-    # first try to ask our IPv4 TFTP server for its boot server
-    bs = tftp.bootServer
-    # if that failed, trying to guess out tftp next server based on the smart proxy hostname
-    bs ||= URI.parse(subnet.tftp.url).host
+    unless subnet.dhcp.has_capability?(:DHCP, :dhcp_filename_hostname)
+      failure _("Unable to determine boot server. The DHCP smart proxy doesn't provide dhcp_filename capability or failed to provide this information.")
+    end
 
     # only smart proxies with V2 API are supported
```

Comment:
Does this comment belong to `unless subnet.dhcp.has_capability?(:DHCP, :dhcp_filename_hostname)`?